pkg/webhookconfig/registration.go

```diff
@@ -739,6 +744,17 @@ func (wrc *Register) updateResourceValidatingWebhookConfiguration(nsSelector map
 		if !ok {
 			return errors.Wrapf(err, "type mismatched, expected map[string]interface{}, got %T", webhooksUntyped[i])
 		}
+
+		currentSelector, _, err := unstructured.NestedMap(webhook, "namespaceSelector")
+		if err != nil {
+			return errors.Wrapf(err, "unable to get validatingWebhookConfigurations.webhooks["+fmt.Sprint(i)+"].namespaceSelector")
+		}
+
+		if reflect.DeepEqual(nsSelector, currentSelector) {
+			wrc.log.V(3).Info("namespaceSelector unchanged, skip updating validatingWebhookConfigurations")
+			continue
+		}
```

Comment:
This is breaking the webhook configurations!

The function this code resides in is responsible for removing the previous webhook configurations and appending new ones in the `webhooks` field. If the current and previous webhook configurations are the same, this will leave us with non-existent `webhooks` field in the webhook configurations, similar to what I'm seeing after this change:
<img width="309" alt="image" src="https://user-images.githubusercontent.com/37282098/154166639-59f684e0-aeef-49f4-a636-c401b36d12ef.png">

I think we should instead perform this check in the function where you're trying to evaluate `nsSelector == nil`: https://github.com/kyverno/kyverno/blob/a970953d513564a50f623ebb4ac3a4351e106d92/pkg/webhookconfig/registration.go#L216
and only then proceed with executing `updateResourceValidatingWebhookConfiguration()` and `updateResourceMutatingWebhookConfiguration()` if there's a change in webhook cfg.